core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -339,4 +349,9 @@ public boolean matchesIndices(String... indices) {
         }
         return false;
     }
+
+    public List<Object> handleTermsLookup(TermsLookup termsLookup) {
+        termsLookup.setIndex((termsLookup.getIndex() == null) ? this.index().name() : termsLookup.getIndex());
```

Comment:
no clone, please. I still think it would be slightly better if we mutate the object in the caller. I guess we need some second opinion...